src/index.js

```diff
@@ -69,13 +58,42 @@ class Message {
       From: encode(this.networkPrefix, this.from),
       Nonce: this.nonce,
       Value: this.value,
-      Method: this.method,
       GasPrice: this.gasPrice,
       GasLimit: this.gasLimit,
+      Method: this.method,
       Params: this.params
     }
     return message
   }
 }
 
+typeCheck = ({ to, from, nonce, value, method, gasPrice, gasLimit }) => {
+  if (typeof nonce !== 'number') throw new Error('Nonce is not a number')
+
+  if (!to) throw new Error('No to address provided')
```

Comment:
While we're at it, can you import filecoin-address here and validate these?

Reply:
The addresses are checked right after this, inside of "newFromString". If it's incorrect, that will handle it.